File: Dddml.Wms.HttpServices.JavaClient/src/org/dddml/wms/domain/UserPermissionMvoStateEventId.java
package org.dddml.wms.domain;

import org.dddml.wms.domain.UserPermissionId;

public class UserPermissionMvoStateEventId
{
    private UserPermissionId userPermissionId;

    public UserPermissionId getUserPermissionId()
    {
        return this.userPermissionId;
    }

    public void setUserPermissionId(UserPermissionId userPermissionId)
    {
        this.userPermissionId = userPermissionId;
    }

    private Long userVersion;

    public Long getUserVersion()
    {
        return this.userVersion;
    }

    public void setUserVersion(Long userVersion)
    {
        this.userVersion = userVersion;
    }

    public String getUserPermissionIdUserId()
    {
        return getUserPermissionId().getUserId();
    }

    public void setUserPermissionIdUserId(String userPermissionIdUserId)
    {
        getUserPermissionId().setUserId(userPermissionIdUserId);
    }

    public String getUserPermissionIdPermissionId()
    {
        return getUserPermissionId().getPermissionId();
    }

    public void setUserPermissionIdPermissionId(String userPermissionIdPermissionId)
    {
        getUserPermissionId().setPermissionId(userPermissionIdPermissionId);
    }


}

File: Dddml.Wms.AdminUI/ts/UserPermissionMvoStateEventId.ts
namespace Dddml.Wms.Domain {

	export class UserPermissionMvoStateEventId {

		private userPermissionId: UserPermissionId;

		getUserPermissionId(): UserPermissionId {
			if(!this.userPermissionId) {
				this.userPermissionId = new UserPermissionId(); 
			}
			return this.userPermissionId;
		}

		setUserPermissionId(userPermissionId: UserPermissionId) {
			this.userPermissionId = userPermissionId;
		}

		private userVersion: string;

		getUserVersion(): string {
			return this.userVersion;
		}

		setUserVersion(userVersion: string) {
			this.userVersion = userVersion;
		}

	}
}

